src/typechecker/FStar.TypeChecker.Util.fs

```diff
@@ -288,7 +288,18 @@ let extract_let_rec_annotation env {lbname=lbname; lbunivs=univ_vars; lbtyp=t; l
                 let body = { body with n = Tm_meta(body', m) } in
                 t, body, recheck
 
-              | Tm_ascribed (_, (Inl t, _), _) -> //no decreases clause here
+              | Tm_ascribed (_, (Inl t, _, use_eq), _) -> //no decreases clause here
```

Comment:
In this case, the type ascription moves to `lbtyp`, since `lbtyp` is only checked via subtyping, we error out here if `use_eq` is set.